docs/examples/physics_using_pybullet/viz_domino.py

```diff
@@ -0,0 +1,224 @@
+
+# Get the position of base and set it.
+base_pos, _ = p.getBasePositionAndOrientation(base)
+base_actor.SetPosition(*base_pos)
+
+
+# Calculate the vertices of the bricks.
+vertices = utils.vertices_from_actor(brick_actor)
+num_vertices = vertices.shape[0]
+num_objects = brick_centers.shape[0]
+sec = np.int(num_vertices / num_objects)
+
+
+"""
+==============
+Syncing Bricks
+==============
+
+Here, we perform three major steps to sync bricks accurately.
+* Get the position and orientation of the bricks from pybullet.
+* Calculate the Rotation Matrix.
+  * Get the difference in orientations (Quaternion).
+  * Generate the corresponding rotation matrix according to that difference.
+  * Reshape it in a 3x3 matrix.
+* Perform calculations to get the required position and orientation.
+* Update the position and orientation.
+"""
+
+def sync_brick(object_index, multibody):
+    pos, orn = p.getBasePositionAndOrientation(multibody)
+
+    rot_mat = np.reshape(
+        p.getMatrixFromQuaternion(
+            p.getDifferenceQuaternion(orn, brick_orns[object_index])),
+        (3, 3))
+
+    vertices[object_index * sec: object_index * sec + sec] = \
+        (vertices[object_index * sec: object_index * sec + sec] -
+         brick_centers[object_index]) @ rot_mat + pos
+
+    brick_centers[object_index] = pos
+    brick_orns[object_index] = orn
+
+
+###############################################################################
+# Here, we define a textblock to display the Avg. FPS and simulation steps.
+
+fpss = np.array([])
+tb = ui.TextBlock2D(text="Avg. FPS: \nSim Steps: ", position=(0, 680),
+                    font_size=30, color=(1, 0.5, 0))
+scene.add(tb)
+
+###############################################################################
+# Set the camera for better visualization.
+
+scene.set_camera(position=(10.46, -8.13, 6.18),
+                 focal_point=(0.0, 0.0, 0.79),
+                 view_up=(-0.27, 0.26, 0.90))
+
+
+###############################################################################
+# Timer callback is created which is responsible for calling the sync and
+# simulation methods.
+
+
+# Create timer callback which will execute at each step of simulation.
+def timer_callback(_obj, _event):
+    global apply_force, fpss
+    cnt = next(counter)
+    showm.render()
+
+    if cnt % 1 == 0:
+        fps = scene.frame_rate
+        fpss = np.append(fpss, fps)
+        tb.message = "Avg. FPS: " + str(np.round(np.mean(fpss), 0)) + \
+                     "\nSim Steps: " + str(cnt)
+
+    # Get the position and orientation of the first brick.
+    brick1_pos, brick1_orn = p.getBasePositionAndOrientation(bricks[0])
+    
+    # Apply force for the first step of simulation.
+    if apply_force:
+        # Apply the force.
+        p.applyExternalForce(bricks[0], -1,
+                             forceObj=[400, 0, 0],
+                             posObj=brick1_pos,
+                             flags=p.WORLD_FRAME)
```

Comment:
Here, you are applying the force at the center of mass of the domino so it will never rotate in this situation. You need to apply the force above the center of mass.